commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java

```diff
@@ -148,6 +176,41 @@ public PortletURL getPortletURL() {
 		return portletURL;
 	}
 
+	public int getRecurringCPDefinitionsCount(
+			CommercePaymentMethodGroupRel commercePaymentMethodGroupRel)
+		throws PortalException {
+
+		if ((commercePaymentMethodGroupRel == null) ||
+			(commercePaymentMethodGroupRel.getGroupId() == 0)) {
+
+			return 0;
+		}
+
+		SearchContext searchContext = new SearchContext();
+
+		Map<String, Serializable> attributes = new HashMap<>();
+
+		attributes.put(Field.STATUS, WorkflowConstants.STATUS_APPROVED);
+
+		attributes.put(CPField.PUBLISHED, true);
+		attributes.put(CPField.SUBSCRIPTION_ENABLED, true);
+
+		long groupId = commercePaymentMethodGroupRel.getGroupId();
```

Comment:
no need for this variable -- just use the getter where it's needed

Reply:
and after that, the `attributes.put` can all be one block

Reply:
This looks a replication of the logic. And still I don't see value in using search over standard finders